### DISCUSSION TITLE
Custom Connection Creators

### DISCUSSION THREAD

**Discussion by luciengaitskell** [sqlalchemy/alembic#941] [Usage Questions] [answered]
With SQLAlchemy it's possible to connect to a server using a custom connection creator like so:
```python 
def getconn():
    conn = connector.connect(
        "project:region:instance",
        "pg8000",
        user="postgres-iam-user@gmail.com",
        db="my_database",
        enable_iam_auth=True,
    )

    return conn

engine = sqlalchemy.create_engine(
    "postgres+pg8000://",
    creator=getconn,
)
```

This enables functionality such as connecting to a Google Cloud SQL server with Automatic IAM using the [cloud-sql-python-connector](https://github.com/GoogleCloudPlatform/cloud-sql-python-connector#how-to-use-this-connector) library (shown above). Is it possible to configure the same in Alembic?

**Comment by zzzeek** [comment, marked as answer]
the create_engine logic is editable within your env.py file that's local to your environment, after running "alembic init".  Go in there and set up the create_engine logic that you need.

in this case the code you would be augmenting is at:

https://github.com/sqlalchemy/alembic/blob/d95e15cac59dc449f712f51183462a8688a51086/alembic/templates/generic/env.py#L59-L64

that is the engine object.   you'd replace this with your create_engine() code above.

https://alembic.sqlalchemy.org/en/latest/tutorial.html#the-migration-environment

**Reply by luciengaitskell** [reply]
Thank you for the quick response. I was able to pass the creator argument right there.

I appreciate the help!
